scenario/mocking.py

```diff
@@ -142,23 +150,37 @@ def _get_secret(self, id=None, label=None):
             except StopIteration:
                 raise SecretNotFoundError()
         else:
+            # if all goes well, this should never be reached. ops.model.Secret will check upon
+            # instantiation that either an id or a label are set, and raise a TypeError if not.
             raise RuntimeError("need id or label.")
 
     @staticmethod
     def _generate_secret_id():
         id = "".join(map(str, [random.choice(list(range(10))) for _ in range(20)]))
         return f"secret:{id}"
 
-    def relation_get(self, rel_id, obj_name, app):
-        relation = self._get_relation_by_id(rel_id)
-        if app and obj_name == self.app_name:
+    def _check_app_data_access(self, is_app: bool):
+        if not isinstance(is_app, bool):
+            raise TypeError("is_app parameter to relation_get must be a boolean")
+
+        if is_app:
+            version = JujuVersion(self._context.juju_version)
+            if not version.has_app_data():
+                raise RuntimeError(
+                    f"setting application data is not supported on Juju version {version}",
+                )
```

Comment:
A tiny change to avoid one indentation level

```suggestion
        if not is_app:
            return
            
        version = JujuVersion(self._context.juju_version)
        if not version.has_app_data():
            raise RuntimeError(
                f"setting application data is not supported on Juju version {version}",
            )
```